bin/server.ml

```diff
@@ -152,6 +152,28 @@ let build_tree_notification (cont : Controller.controller) : Jsonrpc.Notificatio
 
   notifs
 
+let build_watcher_registration_req ~(patterns : GlobPattern.t list) ~(reg_id : string) : unit Lsp.Server_request.t =
+  let watchers = List.map (fun p -> FileSystemWatcher.create ~globPattern:p ()) patterns in
+  let opts = DidChangeWatchedFilesRegistrationOptions.create ~watchers in
+  let opts_json = DidChangeWatchedFilesRegistrationOptions.yojson_of_t opts in
+  let reg = Registration.create ~id:reg_id ~method_:"workspace/didChangeWatchedFiles" ~registerOptions:opts_json () in 
+  Lsp.Server_request.ClientRegisterCapability { registrations = [reg] }
+
+let register_watchers (notify_back : Jsonrpc2.notify_back) (c : Whycode.Controller.controller) =
+  let open Whycode in
+  let open Session_itp in
+  let session = Controller.session c in
+  let files = get_files session in
+  Hfile.iter 
+    (fun _key file ->
+      let path = (system_path session file) in
+      let pat = RelativePattern.create ~baseUri:(`URI (DocumentUri.of_path path)) ~pattern:"*" in
```

Comment:
If I'm reading this correctly, this means that if we have a session at `some/path` we will add a watch for `some/path/*`?

Reply:
Afaik, `get_files` returns the paths of the files in the session, not the session directory. So, we watch only those specific files. The `*` pattern doesn't actually do anything here.